src/float-counts-estimate.cc

```diff
@@ -321,7 +321,7 @@ class FloatCountsEstimator {
     const FloatLmStateWork &work = work_[history_length];
     assert(work.counts.size() == lm_state.counts.size());
 
-    float old_total = lm_state.total,
+    double old_total = lm_state.total,
```

Comment:
you can revert these doubles to floats-- the right fix is what we discussed; but please make that a separate PR from this one.